### ISSUE TITLE
Reset k8s feature remove all non-rancher images

### ISSUE THREAD

**Issue by evertonlperes** [rancher-sandbox/rancher-desktop#701]
## Summary

What happened/what you expected to happen?

**Issue found**
* Resetting k8s through RD settings will remove all  non-rancher pulled images from the system.

**Expected behaviour**
* Reset Kubernetes through RD should keep non-rancher images.

What version of Rancher Desktop are you running?
* `V0.5.0`

What version of Kubernetes are you running?
* `v1.21.5`

Which OS version?
* `macOS 11.6 (20G165) - BigSur / Darwin 20.6.0`

## How to replicate
* Pull sample images using RD `Pull Image` feature or
* Pull sample images with `nerdctl pull apline:latest` or
* Pull sample images with `kim pull node:lts`
* Pulling images with kim, you should be able to see the pulled images on RD image list
* Navigate to K8s Settings and click: `Reset Kubernetes` -> `OK`
* Wait until k8s being reset 
* Verify if pulled images still visible
  * `kim images -a`
  * `On RD, navigate to Images and check the images`

![Screen Shot 2021-09-28 at 2 21 46 PM](https://user-images.githubusercontent.com/37411123/135161373-155a9792-5b0a-4e78-bf85-16d68f7d31a6.png)

## Diagnostics
Resetting k8s through RD UI seems to delete all images non-rancher instead just reset the k8s without removing the non-rancher images.

**Before k8s reset**
Note: manually pulled ubuntu, alpine and ngnix using `kim`

`kim list -a`
```bash
❯ kim images -a
IMAGE                            TAG                    IMAGE ID            SIZE
ubuntu                           latest                 fb52e22af1b01       28.6MB <-- manually pulled
rancher/local-path-provisioner   v0.0.19                148c192562719       42.6MB
rancher/library-busybox          1.32.1                 388056c9a6838       1.45MB
rancher/coredns-coredns          1.8.3                  3885a5b7f138c       43.6MB
rancher/klipper-lb               v0.2.0                 465db341a9e5b       6.46MB
rancher/metrics-server           v0.3.6                 9dd718864ce61       41.2MB
alpine                           latest                 14119a10abf46       2.82MB <-- manually pulled
rancher/pause                    3.1                    da86e6ba6ca19       746kB
rancher/library-traefik          2.4.8                  deaf4b1027ed4       92.7MB
rancher/klipper-helm             v0.6.4-build20210813   f0b5a8f3a50a8       195MB
nginx                            latest                 f8f4ffc8092c9       53.8MB <-- manually pulled
moby/buildkit                    v0.8.3                 cf14c5e88c0eb       56.5MB
rancher/kim                      v0.1.0-beta.6          fef914df5da89       13.8MB
```

After k8s reset
Note: all non-rancher images were deleted.
```bash
IMAGE                            TAG                    IMAGE ID            SIZE
rancher/local-path-provisioner   v0.0.19                148c192562719       42.6MB
rancher/library-busybox          1.32.1                 388056c9a6838       1.45MB
rancher/coredns-coredns          1.8.3                  3885a5b7f138c       43.6MB
rancher/klipper-lb               v0.2.0                 465db341a9e5b       6.46MB
rancher/metrics-server           v0.3.6                 9dd718864ce61       41.2MB
rancher/pause                    3.1                    da86e6ba6ca19       746kB
rancher/library-traefik          2.4.8                  deaf4b1027ed4       92.7MB
rancher/klipper-helm             v0.6.4-build20210813   f0b5a8f3a50a8       195MB
moby/buildkit                    v0.8.3                 cf14c5e88c0eb       56.5MB
rancher/kim                      v0.1.0-beta.6          fef914df5da89       13.8MB
```

**Comment by mattfarina** [comment]
I've reproduced this with nerdctl.

before reset:
```
❯ nerdctl images
WARN[0000] unparsable image name "overlayfs@sha256:6c3699c1cebe59a57f193007cafc6dd21a3288b42b7a3616c09727804f091917"
REPOSITORY    TAG       IMAGE ID        CREATED        SIZE
demo          latest    6c3699c1cebe    3 hours ago    5.9 MiB
nginx         latest    969419c0b7b0    3 hours ago    139.8 MiB
                        6c3699c1cebe    3 hours ago    5.9 MiB
```

After reset of just Kubernetes...
```
❯ nerdctl images
REPOSITORY    TAG    IMAGE ID    CREATED    SIZE
```

**Comment by evertonlperes** [comment]
-> Reopening

Found the issue on Windows:

RD Version: `0.5.0-143-g01465d2`
OS: `Windows 10 PRO 20H2 Azure VM Dv3`

Before soft reset Logs:
```bash

C:\Users\wicke\sandbox>nerdctl pull ubuntu
docker.io/library/ubuntu:latest:                                                  resolved       |++++++++++++++++++++++++++++++++++++++|
index-sha256:a0d9e826ab87bd665cfc640598a871b748b4b70a01a4f3d174d4fb02adad07a9:    done           |++++++++++++++++++++++++++++++++++++++|
manifest-sha256:3555f4996aea6be945ae1532fa377c88f4b3b9e6d93531f47af5d78a7d5e3761: done           |++++++++++++++++++++++++++++++++++++++|
config-sha256:597ce1600cf4ac5f449b66e75e840657bb53864434d6bd82f00b172544c32ee2:   done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:f3ef4ff62e0da0ef761ec1c8a578f3035bef51043e53ae1b13a20b3e03726d17:    done           |++++++++++++++++++++++++++++++++++++++|
elapsed: 3.8 s                                                                    total:  27.2 M (7.2 MiB/s)

C:\Users\wicke\sandbox>nerdctl pull nginx
docker.io/library/nginx:latest:                                                   resolved       |++++++++++++++++++++++++++++++++++++++|
index-sha256:06e4235e95299b1d6d595c5ef4c41a9b12641f6683136c18394b858967cd1506:    done           |++++++++++++++++++++++++++++++++++++++|
manifest-sha256:39065444eb1acb2cfdea6373ca620c921e702b0f447641af5d0e0ea1e48e5e04: done           |++++++++++++++++++++++++++++++++++++++|
config-sha256:f8f4ffc8092c956ddd6a3a64814f36882798065799b8aedeebedf2855af3395b:   done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:4ce73aa6e9b01ef152efc2cb67320b96ba96878ca897167291449f2bd58594f7:    done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:44ac32b0bba890dcb4f6e51224f2ea6044a506f26edf62d39150927efbda7cc8:    done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:8700267f2376da8ddf80c62cb5e1f4183d3bdf6840a84aa87120243aff5259e7:    done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:07aded7c29c6011dfdf02fc98e087c941d3c2661c4e73d134c6491e25231d16c:    done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:91d6e3e593db79f94bb7566733a8253b7bd245cc3ea670f0a349d19fe6960b8a:    done           |++++++++++++++++++++++++++++++++++++++|
layer-sha256:bbe0b7acc89c9d8f880136e22d2243a220895f7a1bfa6113b4e0ebd090e88b81:    done           |++++++++++++++++++++++++++++++++++++++|
elapsed: 4.8 s                                                                    total:  51.3 M (10.7 MiB/s)

C:\Users\wicke\sandbox>nerdctl images
REPOSITORY    TAG       IMAGE ID        CREATED           SIZE
nginx         latest    06e4235e9529    36 seconds ago    139.8 MiB
ubuntu        latest    a0d9e826ab87    44 seconds ago    77.9 MiB

```

After soft reset:
```bash
C:\Users\wicke\sandbox>nerdctl images
REPOSITORY    TAG    IMAGE ID    CREATED    SIZE

C:\Users\wicke\sandbox>
```

Note: All RD default images still available after K8s reset.

**Comment by evertonlperes** [comment]
Adding a few more information related to K8s Reset feature:

On Windows, resetting K8s (soft) I got another error on wsl, but this time related to:
```bash
Creating initial data distribution...
tar: removing leading '/' from member names
agent exited with status 1 signal null
Launching the agent
mount: mounting /mnt/wsl/rancher-desktop/run/data/etc/rancher on /etc/rancher failed: I/O error
WSL: executing: mount -o bind /mnt/wsl/rancher-desktop/run/data/etc/rancher /etc/rancher: Error: wsl.exe exited with code 255
```

Also it seems to crash the entire app: nerdctl, k8s, kim does not work.

Here some steps to reproduce:
* Pull some images using nerdctl
* Navigate to K8s Settings
* Click Reset Kubernetes
* Wait a bit and the error will pop up during the wsl mounting.

Note: Closing and opening the app does not work. 
Workaround: Fixed rebooting the OS or `wsl --shutdown` -> `K8s Reset`

**Logs**
[background.log](https://github.com/rancher-sandbox/rancher-desktop/files/7305004/background.log)
[wsl.log](https://github.com/rancher-sandbox/rancher-desktop/files/7305005/wsl.log)
